UGW: generate config.h for lantiq applications and config.sh for startup scripts.

diff --git a/include/ltq-addons.mk b/include/ltq-addons.mk
new file mode 100644
--- /dev/null
+++ b/include/ltq-addons.mk
@@ -0,0 +1,18 @@
+
+define generate_config_headers
+	$(eval LTQ_CONF_H=$(TMP_DIR)/ifx_config.h)$(eval LTQ_CONF_SH=$(TMP_DIR)/ifx_config.sh) \
+	if [ $(TOPDIR)/.config -nt $(LTQ_CONF_H) -o $(TOPDIR).config -nt $(LTQ_CONF_SH) ]; then \
+		$(SCRIPT_DIR)/config/conf --defconfig=$(TOPDIR)/.config -w $(TOPDIR)/.config Config.in; \
+		echo -en "/*\n * Automatically generated config for C files. Don't edit.\n */\n" > $(LTQ_CONF_H); \
+		grep ^CONFIG_.* .config \
+			| sed 's/=/ /' | awk '{ gsub("-","_",$$1);$$1=toupper($$1);gsub("y$$|m$$","1",$$2);print "#define " $$0 }' \
+			>> $(LTQ_CONF_H); \
+		if [ -f $(LTQ_CONF_H) ]; then mkdir -p $(STAGING_DIR)/usr/include/; cp -f $(LTQ_CONF_H) $(STAGING_DIR)/usr/include/; fi; \
+		echo -en "#\n# Automatically generated config for startup scripts. Don't edit.\n#\n" > $(LTQ_CONF_SH); \
+		grep ^#define $(LTQ_CONF_H) \
+			| sed -e 's/#define //' -e 's/ /=/' \
+			| awk '{ print "export "$$0 }' \
+			>> $(LTQ_CONF_SH); \
+	fi
+endef
+
diff --git a/include/subdir.mk b/include/subdir.mk
--- a/include/subdir.mk
+++ b/include/subdir.mk
@@ -12,6 +12,8 @@ else
   SUBTARGETS:=clean download prepare compile install update refresh prereq dist distcheck configure
 endif
 
+include include/ltq-addons.mk
+
 subtarget-default = $(filter-out ., \
 	$(if $($(1)/builddirs-$(2)),$($(1)/builddirs-$(2)), \
 	$(if $($(1)/builddirs-default),$($(1)/builddirs-default), \
@@ -64,6 +66,8 @@ define stampfile
 	@mkdir -p $$$$(dirname $$($(1)/stamp-$(3)))
 	@touch $$($(1)/stamp-$(3))
 
+	@$$(call generate_config_headers)
+
   $$(if $(call debug,$(1),v),,.SILENT: $$($(1)/stamp-$(3)))
 
   .PRECIOUS: $$($(1)/stamp-$(3)) # work around a make bug
